validator/validator.go

```diff
@@ -580,3 +580,11 @@ func LdapFilter(value interface{}, _ cty.Path) diag.Diagnostics {
 
 	return diags
 }
+
+// StringInSlice returns a SchemaValidateFunc which tests if the provided value
+// is of type string and matches the value of an element in the valid slice
+// will test with in lower case if ignoreCase is true.
+// Modified version eliminates the need of wrapping for ValidateDiagFunc.
+func StringInSlice(ignoreCase bool, strings ...string) schema.SchemaValidateDiagFunc {
```

Comment:
Let's keep the same ordering of the args as the original `StringInSlice`.

Reply:
Can you move this next to the other validation funcs from TF module?

Reply:
nvm about the args ordering, you are using variadic func...

Reply:
I can move it, but can't change the order of args, it's a Go rule - `strings ...string` must be the last argument.

Reply:
So what I did with the other funcs is actually copy the code from TF repo and re-implement them to be a `SchemaValidateDiagFunc`, instead of just wrapping them.